File: day1/day1_2.py
def day1_2():
    left = []
    right = []
    result = []

    with open("day1/input.txt", "r") as file:
        for line in file:
            values = line.split()
            if len(values) == 2:
                left.append(int(values[0]))
                right.append(int(values[1]))

    for l in left:
        count = 0
        for r in right:
            if l == r:
                count += 1
        result.append(l * count)

    print(sum(result))

File: day1/day1_2.go
package Day1

import (
	"os"
	"strconv"
	"strings"
)

func sumInsideSlice(v []int) int {
	sum := 0
	for _, value := range v {
		sum += value
	}
	return sum
}

func Day1_2() int {
	var left, right, result []int

	data, _ := os.ReadFile("day1/input.txt")
	lines := strings.Split(string(data), "\n")

	for _, line := range lines {
		parts := strings.Fields(line)
		if len(parts) == 2 {
			l, _ := strconv.Atoi(parts[0])
			left = append(left, l)
			r, _ := strconv.Atoi(parts[1])
			right = append(right, r)
		}
	}

	for _, l := range left {
		count := 0
		for _, r := range right {
			if l == r {
				count++
			}
		}
		result = append(result, l*count)
	}

	return sumInsideSlice(result)
}
